android/java/org/chromium/chrome/browser/app/domain/CryptoModel.java

```diff
@@ -330,10 +315,13 @@ public void updateCoinType(Integer coin, Callback1<Integer> callback) {
     }
 
     public void updateCoinType(Callback1<Integer> callback) {
-        mBraveWalletService.getSelectedCoin(coinType -> {
-            _mCoinTypeMutableLiveData.postValue(coinType);
+        mKeyringService.getAllAccounts(allAccounts -> {
+            // Current coin is the coin of selected account.
+            @CoinType.EnumType
+            int coin = allAccounts.selectedAccount.accountId.coin;
+            _mCoinTypeMutableLiveData.postValue(coin);
             if (callback != null) {
-                callback.call(coinType);
+                callback.call(coin);
```

Comment:
Android UI relies on concept of 'current coin' which should be replaced with 'current account' but this would be too much for this PR